cmake/adjust_global_compile_flags.cmake

```diff
@@ -279,6 +279,11 @@ if (MSVC)
     set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} /Gw /GL")
     set(CMAKE_CXX_FLAGS_MINSIZEREL "${CMAKE_CXX_FLAGS_MINSIZEREL} /Gw /GL")
   endif()
+
+  # DLL initialization errors due to old conda msvcp140.dll dll are a result of the new MSVC compiler
+  # See https://developercommunity.visualstudio.com/t/Access-violation-with-std::mutex::lock-a/10664660#T-N10668856
+  # Remove this definition once the conda msvcp140.dll dll is updated.
+  add_compile_definitions(_DISABLE_CONSTEXPR_MUTEX_CONSTRUCTOR)
```

Comment:
This switches ABIs.  All the code in the same binary(EXE or DLL) should be built with the same ABI, which means someone may want it while someone else does not want it.  So, I suggest not setting it here.  We may put it at https://github.com/microsoft/onnxruntime/blob/main/tools/ci_build/build.py#L1554 , which only have effects on our pipelines.